libethereum/State.cpp

```diff
@@ -541,6 +541,7 @@ std::pair<ExecutionResult, TransactionReceipt> State::execute(EnvInfo const& _en
 	u256 const startGasUsed = _envInfo.gasUsed();
 	executeTransaction(e, _t, onOp);
 
+	res.changeLog = m_changeLog;
```

Comment:
I don't like it either. This must go to `dev::test` namespace unless proven it is not possible to be done this way.

And big no to build flags. Let's find a way to do it nicely without affecting not-test related code.